shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -490,6 +503,59 @@ - (void)dealloc {
   [super dealloc];
 }
 
+#pragma mark - Status Bar touch event related swizzling
+
+constexpr CGFloat kStandardStatusBarHeight = 20.0;
+
+static IMP __original_sendevent = nil;
+
+void _flutter_sendevent(id self, SEL _cmd, UIEvent *event) {
+  // Call the original implementation.
+  ((void(*)(id,SEL,UIEvent*))__original_sendevent)(self, _cmd, event);
+
+  // Intercept touchBegan events.
+  UITouch *touch = event.allTouches.anyObject;
+  if (touch.phase != UITouchPhaseBegan) {
+    return;
+  }
+
+  // Don't handle the touch for double-height status bars. iOS will intercept
+  // and open the associated app.
+  UIApplication *app = [UIApplication sharedApplication];
+  CGRect statusBarFrame = app.statusBarFrame;
+  if (statusBarFrame.size.height != kStandardStatusBarHeight) {
+    return;
+  }
+
+  // Ignore taps outside the status bar. The view already handles them.
+  UIViewController *viewController = app.delegate.window.rootViewController;
+  CGPoint location = [touch locationInView:viewController.view.window];
+  if (!CGRectContainsPoint(statusBarFrame, location)) {
+    return;
+  }
+
+  // Issue touch start and end events: as of iOS 10.2, we get get no
+  // touchEnded event for status bar taps.
+  [viewController touchesBegan:[event allTouches] withEvent:event];
```

Comment:
After further playing with this, it turns out whether you get touch ended events depends on where you listen.

When overriding `touchesBegan` and `touchesEnded` methods, you only get the began event (at the *end* of a tap), but not the ended event, *unless* there was a drag involved. 

When overriding or swizzling `UIApplication`'s `sendEvent` you get both `began` and `ended` events at the correct moments.